stdlib/Dates/src/io.jl

```diff
@@ -529,6 +529,27 @@ end
 
 # show
 
+for T in (
+    :Year, :Month, :Week, :Day,
+    :Hour, :Minute, :Second, :Millisecond, :Microsecond, :Nanosecond
+)
+    @eval function Base.show(io::IO, p::$T)
+        if get(io, :compact, false)
+            print(io, p)
+        else
+            print(io, $(string(T)), '(', p.value, ')')
+        end
+    end
+end
+
+function Base.show(io::IO, ::MIME"text/plain", p::Period)
+    print(io, p)
+end
+
+function Base.print(io::IO, p::Period)
+    print(io, string(p))
```

Comment:
`string` is never needed inside of `print`, as it's defined as (effectively) `sprint(print, x)`. So this can be just `print(io, p)`. Also, the separate method for plain text shouldn't be necessary.

Reply:
ah right. ok

Reply:
Oh you're right. Is the definition here necessary or does the fallback look as you'd expect?

Reply:
Guess that depends on how we want it. Without that print command that calls string, then show(x), print(x), repr(x) all show "Dates.Hour(1)" with x = Dates.Hour(1), and then string(x) is "1 hour"

Reply:
with that print call that calls string, then string(x) and print(x) would show "1 hour"